code/tsp_ls1.py

```diff
@@ -57,20 +55,30 @@ def calculate_route_cost(route,cost_matrix):
     return cost
 
 
-def two_opt(route_matrix,dist_matrix,cost):
-    bestroute = route_matrix
-    improvement = False
-    # TODO: This loop does not consider flips which include endpoints
-    for i in range(1,len(route_matrix)-1):
-        for k in range(i+i,len(route_matrix)):
-            improved_route = route_matrix[:]
-            improved_route[i:k] = route_matrix[k-1:i-1:-1] #Here we will take the section to flip and flip it and replace the order.
-            new_cost = calculate_route_cost(improved_route,dist_matrix)
-            if new_cost < calculate_route_cost(bestroute,dist_matrix):
-                bestroute = improved_route
-                improvement = True
-    route_matrix = bestroute
-    return improvement, route_matrix, calculate_route_cost(route_matrix,dist_matrix)
+def two_opt(dist_matrix,nodes,start_nodes,tracepoint_pipe : Pipe, solution_pipe : Pipe):
+    initial = 1
+    start_time = time.process_time()
+    mat = distance_calculator(nodes)
+    while 1:
+        for x in start_nodes:
+            greedy_route,greedy_total_cost = greedy_heuristic(mat,x)
+            route_matrix = greedy_route
+            if initial == 1:
```

Comment:
Can you just initialize bestroute to None and bestcost to infinity, and get rid of the check for initial?

Reply:
Yeah idk why I did that

Reply:
yes did it god call